zaza/model.py

```diff
@@ -1211,18 +1213,24 @@ async def async_run_action_on_units(units, action_name, action_params=None,
 
     model = await get_model(model_name)
     actions = []
+    async_actions = []
     for unit_name in units:
         unit = await async_get_unit_from_name(unit_name, model)
         action_obj = await unit.run_action(action_name, **action_params)
-        actions.append(action_obj)
+        if inspect.isawaitable(action_obj):
```

Comment:
All actions returned from unit.run_action should be awaitable - both in juju 2.9 and 3.x